### PR TITLE
Fix MME submissions page crashing

### LINKED ISSUE #4214: Broken Matchmaker submission page
This happens when you have a case with variants in MME and you click on submission details:

image

Error:

```
Traceback (most recent call last):
  File "/venv/lib/python3.8/site-packages/flask/app.py", line 2529, in wsgi_app
    response = self.full_dispatch_request()
  File "/venv/lib/python3.8/site-packages/flask/app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/venv/lib/python3.8/site-packages/flask_cors/extension.py", line 176, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/venv/lib/python3.8/site-packages/flask/app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "/venv/lib/python3.8/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/worker/app/scout/server/utils.py", line 86, in decorated_function
    context = func(*args, **kwargs)
  File "/home/worker/app/scout/server/blueprints/cases/views.py", line 132, in matchmaker_matches
    data = controllers.matchmaker_matches(request, institute_id, case_name)
  File "/home/worker/app/scout/server/blueprints/cases/controllers.py", line 1319, in matchmaker_matches
    pat_matches = parse_matches(patient_id, server_resp["content"]["matches"])
  File "/home/worker/app/scout/parse/matchmaker.py", line 180, in parse_matches
    mdate = datetime.datetime.fromtimestamp(milliseconds_date / 1000.0)
TypeError: unsupported operand type(s) for /: 'str' and 'float'
```

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
Fix #4214 

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Deploy on stage and using both patientmatcher 4.4 and patientmatcher 4.5:
2. Go ti any case and pin one or more variants
3. Submit them to Matchmaker
4. Go to submission page details

**Expected outcome**:
- Main branch should crash with patientmatcher 4.5
- This branch should work

**Review:**
- [x] code approved by DNIL
- [x] tests executed by CR

### CODE REVIEW

**Review by dnil — APPROVED** [review]
Type agnostic language they said.. 😸

**Comment by northwestwitch** [comment]
> Type agnostic language they said.. 😸

Thanks for the review, but it doesn't work. I also want to understand why it stopped working and if it has to do with how dates are saved in the latest mongo versions, because that could have happened and could have potential repercussions elsewhere, I'll mark this PR as ready for review as soon as I've tested a bit more!

**Comment by northwestwitch** [comment]
Yes. The old mongodb 4.4 returned the date in milliseconds (tested locally with patentmatcher 4.4 docker-compose)
Then the scout page work as expected:

image

When I switch to docker-compose patientmatcher 4.5 (the latest) I get dates as strings instead:

image

File: scout/parse/matchmaker.py [hunk 1]
```diff
@@ -175,9 +175,11 @@ def parse_matches(patient_id, match_objs):
     parsed_matches = []
 
     for match_obj in match_objs:
-        # convert match date from millisecond to readable date
-        milliseconds_date = match_obj["created"]["$date"]
-        mdate = datetime.datetime.fromtimestamp(milliseconds_date / 1000.0)
+        match_date = match_obj["created"]["$date"]
+        try:  # Patientmatcher < 4.5 returns dates in milliseconds
+            mdate = datetime.datetime.fromtimestamp(match_date / 1000.0)
+        except TypeError:  # Patientmatcher >= 4.5 returns dates in isoformat
```
Comment (northwestwitch) [inline scout/parse/matchmaker.py:181]: Works,tested locally with patientmatcher 4.5

image

**Comment by northwestwitch** [comment]
Tested fine also on cg-vm1 with this case: https://scout-stage.scilifelab.se/cust000/F0006358-mip7

image

File: scout/parse/matchmaker.py (hunk 1 above)
Comment (dnil) [inline scout/parse/matchmaker.py:181]: 👍

**Pull request merged**